ddtrace/appsec/handlers.py

```diff
@@ -169,32 +169,43 @@ def _on_django_func_wrapped(fn_args, fn_kwargs, first_arg_expected_type):
     # path parameters)
     if _is_iast_enabled() and fn_args and isinstance(fn_args[0], first_arg_expected_type):
         from ddtrace.appsec.iast._taint_tracking import OriginType  # noqa: F401
+        from ddtrace.appsec.iast._taint_tracking import is_pyobject_tainted
         from ddtrace.appsec.iast._taint_tracking import taint_pyobject
         from ddtrace.appsec.iast._taint_utils import LazyTaintDict
 
-        if not isinstance(fn_args[0].COOKIES, LazyTaintDict):
-            fn_args[0].COOKIES = LazyTaintDict(fn_args[0].COOKIES, origins=(OriginType.COOKIE_NAME, OriginType.COOKIE))
-        if not isinstance(fn_args[0].GET, LazyTaintDict):
-            fn_args[0].GET = LazyTaintDict(fn_args[0].GET, origins=(OriginType.PARAMETER_NAME, OriginType.PARAMETER))
-        if not isinstance(fn_args[0].POST, LazyTaintDict):
-            fn_args[0].POST = LazyTaintDict(fn_args[0].POST, origins=(OriginType.BODY, OriginType.BODY))
-        if not isinstance(fn_args[0].META, LazyTaintDict):
-            fn_args[0].META = LazyTaintDict(fn_args[0].META, origins=(OriginType.HEADER_NAME, OriginType.HEADER))
-        if not isinstance(fn_args[0].headers, LazyTaintDict):
-            fn_args[0].headers = LazyTaintDict(fn_args[0].headers, origins=(OriginType.HEADER_NAME, OriginType.HEADER))
-        fn_args[0].path = taint_pyobject(
-            fn_args[0].path, source_name="path", source_value=fn_args[0].path, source_origin=OriginType.PATH
+        http_req = fn_args[0]
+
+        if not isinstance(http_req.COOKIES, LazyTaintDict):
+            http_req.COOKIES = LazyTaintDict(http_req.COOKIES, origins=(OriginType.COOKIE_NAME, OriginType.COOKIE))
+        if not isinstance(http_req.GET, LazyTaintDict):
+            http_req.GET = LazyTaintDict(http_req.GET, origins=(OriginType.PARAMETER_NAME, OriginType.PARAMETER))
+        if not isinstance(http_req.POST, LazyTaintDict):
+            http_req.POST = LazyTaintDict(http_req.POST, origins=(OriginType.BODY, OriginType.BODY))
+        if not is_pyobject_tainted(getattr(http_req, "_body", None)):
```

Comment:
Is this if independent from the previous one? or maybe we can use an `elif` here?

Reply:
they are all independent and taint different kinds of information.